addons/crm/models/res_partner.py

```diff
@@ -52,6 +52,19 @@ def _compute_opportunity_count(self):
                     partner.opportunity_count += group['partner_id_count']
                 partner = partner.parent_id
 
+    @api.onchange('parent_id', 'company_type')
+    def _onchange_parent_id_for_team_id(self):
+        # While creating / updating parent, take the parent salesteam value and set on current record
+        if not self.team_id and self.parent_id.team_id and self.company_type == 'person':
+            self.team_id = self.parent_id.team_id
+
+    def _fields_sync(self, values):
+        # sync salesteam with parent
```

Comment:
Could be explained a bit more and put in docstring (see other docstring example in my other comment).